### ISSUE TITLE
Doesn't squish on Windows

### ISSUE THREAD

**Issue by anpin** [control4/drivers-driverpackager#8]
Hi, if I set squishLua to true and run the script like this
```python .\scripts\drivers-driverpackager\dp3\driverpackager.py -v -ae .\src\ .\bin\ 'My Driver Project.c4zproj``` 
It gives me this 
```2021-05-31-21:10:44: DriverPackager: Error squishing lua [WinError 2] The system cannot find the file specified```
I can build it without squishing using the script and I can build it in DriverEditor with squish, although DriverEditor is really really slow for some reason.

**Comment by WWalshC4** [comment]
Do you have a `squishy` file in the same directory as the `My Driver Project.c4zproj` file?

**Comment by anpin** [comment]
Hi @WWalshC4  thanks for your swift reply. Indeed the file was not there however I remember putting it in. I'll try building it again later this week.

**Comment by anpin** [comment]
I was finally able to get back to it and here is what I found.
I have following directory structure:
my-module
├── scripts
│   └── drivers-driverpackager <- this repo as git submodule
├── src <- folder with actual code
│   ├── my-driver.c4zproj
│   ├── squishy
│   ├── driver.lua
│   ├── driver.xml
│   └── some_other_module.lua
└─ build.ps1 <- one line script with the command from original post
I've added `C:\Program Files (x86)\Control4\DriverEditor301` to PATH so I can use lua.exe which is bundled with the DriverEditor.
I'm able to squish the driver if I run `lua.exe .\scripts\drivers-driverpackager\dp3\squish --no-minify .\src\` however running `python .\scripts\drivers-driverpackager\dp3\driverpackager.py -v -ae .\src\ .\bin\ 'my-driver.c4zproj` returns same error as in the original post `2021-08-30-19:09:28: DriverPackager: Error squishing lua [WinError 2] The system cannot find the file specified`

**Comment by WWalshC4** [comment]
Can you provide a zipfile/link to repo for the project?  Feel free to strip all code out of any Lua files that you wish to keep proprietary.
Have you tried running the same python scripts on a WSL/Cygwin instance to see if they work under *nix?

**Comment by anpin** [comment]
just tried running it on linux and it doesn't build at all because of path separator differences between platforms ( \ vs /   kind of problem)

**Comment by WWalshC4** [comment]
As requested, can you provide the source of the .c4zproj, the squishy and enough of the other components to get some context on what is being done?  Without source, I'm not going to be able to help you.

**Comment by anpin** [comment]
[repro.tar.gz](https://github.com/control4/drivers-driverpackager/files/7129148/repro.tar.gz)
please find in attach

**Comment by WWalshC4** [comment]
Process for testing:

- I'm on linux, so I modified your directory references to use the `/` dir separator.  I also modified the references that began with '../' to be './../' - mostly because that's the standard I use.  I doubt this had any effect though.

I then tried running your .c4zproj directly with driverpackager.py, and got a driver output as expected.  I also saw as output from the script:

```
2021-09-08-15:03:21: string indices must be integers
The terminal process "bash '-c', 'driverpackager.py -v ./ ./../ 'AMX PRECIS DSP 8x8.c4zproj''" failed to launch (exit code: 255).
```

Digging in on this, I think this is because you're trying to add the entire `../shared` directory using the `Item` to include it, but you're also adding every file from that directory in your `squishy` and in your `<Squishy>` `manualsquish` section.  This error is, I think, because there aren't any files left in that dir to add that haven't already been included in the squished output.  Excluding that directory from being added removed the error, although the output was the same - a .c4z containing two files - a squished driver.lua and a driver.xml.

I think your problem on Windows is stemming from using `lua.exe` instead of `luajit.exe` - I get errors on Linux when running squish with lua5.2 instead of luajit, and this is what the `driverpackager.py` script calls, so you'd be best placed to use luajit.

TL;DR - use luajit if you're using squish, and don't include a directory of files through the `.c4zproj` that you're also including files from in your `squishy`.

**Comment by anpin** [comment]
Thanks for in-depth explanation. This directory/project structure is just a first one I managed to make work with the DriverEditor on WIndows to share code between two modules with different I/O count and pack it all into single file which runs.  Not sure why are you getting "AMX PRECIS DSP 8x8.c4zproj" since I replaced all spaces with _ as they were not playing nice with Github Actions. Doubt that I ever going to pursue making the pipeline or script to work on this particular project as driver seems to work, client is satisfied and they don't want me to publish it open source, but I will use that knowledge with the next one, so thank you and we can close the issue now. (Also finally got my Linux environment setup and will be using it from now on for developing lua drivers)
